app/src/main/java/org/stepic/droid/code/ui/CodeAnalyzer.kt

```diff
@@ -0,0 +1,104 @@
+package org.stepic.droid.code.ui
+
+import org.stepic.droid.util.countWhile
+import org.stepic.droid.util.substringOrNull
+
+/**
+ * Class for smart code analyzing
+ */
+object CodeAnalyzer {
+    private const val LINE_BREAK = '\n'
+
+    private val brackets = hashMapOf(
```

Comment:
Use Char to Char

Reply:
tbh, I think that using of hashMap here is overkill, but it contains only 3 items, so it doesn't matter